boskos/janitor/janitor.go

```diff
@@ -0,0 +1,87 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"os/exec"
+	"time"
+
+	"github.com/Sirupsen/logrus"
+	"k8s.io/test-infra/boskos/client"
+)
+
+type semaphore chan bool
+
+// acquire 1 resources
+func (s semaphore) P() {
+	s <- true
+}
+
+// release 1 resources
+func (s semaphore) V() {
+	<-s
+}
+
+var (
+	clean       = janitorClean
+	janitorPool = make(semaphore, 10)
+)
+
+func main() {
+	logrus.SetFormatter(&logrus.JSONFormatter{})
+	boskos := client.NewClient("Janitor", "http://boskos")
+	logrus.Info("Initialized boskos client!")
+	for {
+		if proj, err := boskos.Acquire("project", "dirty", "cleaning"); err != nil {
```

Comment:
Much cleaner! 

There is one more improvement I would like to make: we should not acquire the 11th project until we're done cleaning 1 of the first 10. This is cleaner, and gives us the freedom to create multiple janitor pods if necessary.

What do you think about something along these lines:

```
func main() {
  poolSize := 10
  pool := make(semaphore, poolSize)
  for i:=0;i<poolSize;i++ { go janitor(boskos, pool) }
  for {
    p, err := boskos.Acquire()
    if err != nil { sleep }
    pool <- p  // will block until the pool has a free slot.
  }
}

func janitor(boskos, pool) {
  for {
    proj := <- pool  // will block until there is an item in the pool
    janitorClean(proj)
  }
}
```